src/components/query-panel/QueryPanelCtrl.ts

```diff
@@ -19,20 +19,28 @@ export class QueryPanelCtrl implements ng.IComponentController {
 
   executeAction(): void {
     this.executionActionHelper().then((response: any) => {
-      this.statusCode = 200;
+      if (isNotNullOrUndefined(response.status)) {
```

Comment:
The response should never be null here. If `executionActionHelper` goes into the default switch statement, we should reject the promise as well as writing an error. That way this part of the code will never be executed with a null response object. Furthermore, is it every possible for a response object not to have a status? I assume that in the event of a server error a status of `5xx` would be returned at least.